### PR TITLE
chore: pass the private data to the runner via config

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright/src/common/config.ts [hunk 1]
```diff
@@ -96,6 +97,11 @@ export class FullConfigInternal {
       workers: 0,
       webServer: null,
     };
+    for (const key in config) {
+      if (key.startsWith('@'))
+        (this.config as any)[key] = (config as any)[key];
```
Comment (dgozman) [inline packages/playwright/src/common/config.ts:102]: I wonder whether we `JSON.stringify()` the config somewhere, which would probably break?

File: packages/playwright/src/runner/runner.ts [hunk 2]
```diff
@@ -48,10 +51,11 @@ export class Runner {
     this._config = config;
   }
 
-  async listTestFiles(projectNames: string[] | undefined, projectGrep: string | undefined): Promise<any> {
+  async listTestFiles(frameworkPackage: string | undefined, projectNames: string[] | undefined, projectGrep: string | undefined): Promise<any> {
```
Comment (dgozman) [inline packages/playwright/src/runner/runner.ts:54]:
```suggestion
  async listTestFiles(frameworkPackage: string | undefined, projectNames: string[] | undefined, projectGrep: string | undefined): Promise<ConfigListFilesReport> {
```

File: packages/playwright/src/runner/runner.ts [hunk 3]
```diff
@@ -39,6 +40,8 @@ type ProjectConfigWithFiles = {
 
 type ConfigListFilesReport = {
   projects: ProjectConfigWithFiles[];
+  cliEntryPoint?: string;
+  error?: TestError;
```
Comment (dgozman) [inline packages/playwright/src/runner/runner.ts:44]: Where is this one set?

**Review by dgozman — APPROVED** [review]

File: packages/playwright/src/common/config.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright/src/common/config.ts:102]: I hope we never do that

File: packages/playwright/src/runner/runner.ts (hunk 3 above)
Comment (pavelfeldman) [reply packages/playwright/src/runner/runner.ts:44]: withRunnerAndMutedWrite

**Pull request merged**